server/src/main/java/io/spine/server/bus/Bus.java

```diff
@@ -141,15 +146,37 @@ public final void post(Iterable<T> messages, StreamObserver<Ack> observer) {
         checkNotNull(messages);
         checkNotNull(observer);
 
-        Iterable<T> filteredMessages = filter(messages, observer);
+        StreamObserver<Ack> wrappedObserver = wrappedObserver(messages, observer);
+        filterAndPost(messages, wrappedObserver);
+    }
+
+    private void filterAndPost(Iterable<T> messages, StreamObserver<Ack> observer) {
+        Collection<T> filteredMessages = filter(messages, observer);
         if (!isEmpty(filteredMessages)) {
             store(filteredMessages);
-            Iterable<E> envelopes = transform(filteredMessages, toEnvelope());
+            Iterable<E> envelopes = filteredMessages.stream()
+                                                    .map(this::toEnvelope)
+                                                    .collect(toList());
             doPost(envelopes, observer);
         }
         observer.onCompleted();
     }
 
+    /**
+     * Transforms the given {@link StreamObserver}.
```

Comment:
With the given method signature this first sentence explains nothing. Please let the reader follow your real reason behind this method creation.